File: app/src/main/java/io/mainframe/hacs/ssh/DoorCommand.java
package io.mainframe.hacs.ssh;

import io.mainframe.hacs.main.Status;

/**
 * Created by holger on 11.08.16.
 */
public final class DoorCommand {

    private final String command;

    private DoorCommand(String command) {
        this.command = command;
    }

    public static DoorCommand getSwitchDoorStateCmd(Status status) {
        return new DoorCommand("set-status " + status.getMqttValue());
    }

    public static DoorCommand getInnerGlassDoorBuzzerCmd() {
        return new DoorCommand("open-door glass");
    }

    public static DoorCommand getInnerMetalDoorBuzzerCmd() {
        return new DoorCommand("open-door main");
    }

    public static DoorCommand getOuterDoorBuzzerCmd() {
        return new DoorCommand("open-door downstairs");
    }

    public String get() {
        return this.command;
    }

    @Override
    public String toString() {
        return String.format("DoorCommand{command='%s'}", command);
    }
}

File: app/src/main/java/io/mainframe/hacs/ssh/DoorCommand.kt
package io.mainframe.hacs.ssh

import io.mainframe.hacs.main.Status

/**
 * Created by holger on 11.08.16.
 */
class DoorCommand private constructor(private val command: String) {
    fun get(): String {
        return this.command
    }

    override fun toString(): String {
        return "DoorCommand(command='$command')"
    }


    companion object {
        fun getSwitchDoorStateCmd(status: Status): DoorCommand {
            return DoorCommand("set-status " + status.mqttValue)
        }

        @JvmField
        val innerGlassDoorBuzzerCmd = DoorCommand("open-door glass")

        @JvmField
        val innerMetalDoorBuzzerCmd = DoorCommand("open-door main")

        @JvmField
        val outerDoorBuzzerCmd = DoorCommand("open-door downstairs")
    }
}
